### PR TITLE
Create /DELETE route for Words - resolves ijemmao/igbo_api#131

### LINKED ISSUE #131: Create /DELETE route for Words
Currently, there are some duplicate words in the database, but there's no way to remove those words.

Create a new /DELETE route that a merger would use to delete a word document from the production database.

The route to implement would be DELETE `api/v1/edit/words/:id` which will allow for a single word document to be deleted.

### PR DESCRIPTION
**Pull request by oemekaogala**
In this patch resolves #131. 

I completed two things:

1. the reorganized the documentation comments to align with the corresponding route. This will make it easier for anyone to understand the relationship between the comment/swagger documentation. While going through this documentation, I also noticed there was a missing GET /examples route. The function was already implemented, so I also added this route.

2. Added a new route `api/v1/edit/words/:id` to delete a word from mongodb. I also included documentation for this new route.

**Note**: I decided to return the word as a response when it is successfully deleted

### CODE REVIEW

File: src/controllers/words.js [hunk 1]
```diff
@@ -199,3 +199,22 @@ export const putWord = (req, res) => {
       return res.send({ error: 'An error has occurred while updating the word, double check your provided data.' });
     });
 };
+
+export const deleteWord = (req, res) => {
+  const { params: { id } } = req;
```
Comment (ijemmao) [inline src/controllers/words.js:204]: There's a chance that a `WordSuggestion` and `GenericWord` document might reference to a `Word` document with either the `originalWordId` or even the `merged` key. I think we should also make a pass through those documents and remove references from those objects.

File: src/routers/router.js [hunk 2]
```diff
@@ -5,96 +5,105 @@ import { getExamples, getExample } from '../controllers/examples';
 const router = express.Router();
```
Comment (ijemmao) [inline src/routers/router.js:5]: It looks like you didn't create an express route for deleting words yet. Inside `editorRouter.js` could you create the delete route like `editorRouter.delete('/words/:id')`?

**Review by ijemmao — CHANGES_REQUESTED** [review]
This is shaping up! Just a couple things:

* Remove all references to the recently deleted `Word` document inside `WordSuggestion` and `GenericWord` documents
* Create a new express route in `editorRouter` to actually delete a word
* Add Swagger docs to that new route
* Create tests to make sure that not only is the `Word` document actually deleted but there are no more references to that word

**Comment by oemekaogala** [comment]
Okay thanks for the comments. Will work on this

**Comment by ijemmao** [comment]
Closing this PR since it's been inactive. Reopen this PR once you are ready for a new review.

**Pull request closed without merging**